File: user-service/src/main/java/com/ttalkak/user/common/util/Json.java
package com.ttalkak.user.common.util;

import com.nimbusds.jose.shaded.gson.Gson;

public class Json {
    private static final Gson gson = new Gson();
    public static <T> String serialize(T object) {
        return gson.toJson(object);
    }

    public static <T> T deserialize(String json, Class<T> clazz) {
        return gson.fromJson(json, clazz);
    }
}

File: compute-service/src/main/kotlin/com/ttalkak/compute/common/util/Json.kt
package com.ttalkak.compute.common.util

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

object Json {
    fun <T> serialize(instance: T): String {
        return Gson().toJson(instance)
    }

    fun <T> deserialize(json: String, clazz: Class<T>): T {
        return Gson().fromJson(json, clazz)
    }

    inline fun <reified T> deserialize(json: String): T {
        val gson = Gson()
        val type = object : TypeToken<T>() {}.type
        return gson.fromJson(json, type)
    }
}